src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -62,12 +62,34 @@ public void createJavaDocComment_sampleCode() {
         JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
     String expected =
         "sample codes:\n"
-            + "<pre><code>\n"
+            + "<pre>{@code\n"
             + "resource = project/{project}/shelfId/{shelfId}\n"
-            + "</code></pre>";
+            + "}</pre>";
     assertEquals(javaDocComment.comment(), expected);
   }
 
+  @Test
+  public void createJavaDocComment_sampleCodePreserveIndentAndLineBreaks() {
+    String comment = "sample codes:";
+    String sampleCode1 =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings.newBuilder().setEndpoint(myEndpoint).build();\n";
+    String sampleCode2 =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
```

Comment:
Can we add back the "badly-indented" test case, and replace this one? Up to you whether to keep `sampleCode1`.